src/plugins/func/funcUtils.ts

```diff
@@ -0,0 +1,77 @@
+import yaml from "js-yaml";
+import Serverless from "serverless";
+import httpBinding from "./bindingTemplates/http.json"
+
+export class FuncPluginUtils {
+
+  public static getServerlessYml(sls: Serverless) {
+    return sls.utils.readFileSync("serverless.yml");
+  }
+
+  public static getFunctionsYml(sls: Serverless, serverlessYml?: any) {
+    serverlessYml = serverlessYml || FuncPluginUtils.getServerlessYml(sls);
+    return serverlessYml["functions"];
+  }
+
+  public static updateFunctionsYml(sls: Serverless, functionYml: any, serverlessYml?: any) {
+    serverlessYml = serverlessYml || FuncPluginUtils.getServerlessYml(sls);
+    serverlessYml["functions"] = functionYml;
+    sls.utils.writeFileSync("serverless.yml", yaml.dump(serverlessYml));
+  }
+
+  public static getFunctionHandler(name: string) {
+    return `"use strict";
+
+module.exports.handler = async function (context, req) {
+  context.log("JavaScript HTTP trigger function processed a request.");
+
+  if (req.query.name || (req.body && req.body.name)) {
+    context.res = {
+      // status: 200, /* Defaults to 200 */
+      body: "${name} " + (req.query.name || req.body.name)
+    };
+  }
+  else {
+    context.res = {
+      status: 400,
+      body: "Please pass a name on the query string or in the request body"
+    };
+  }
+};`
+  }
+
+  public static getFunctionJsonString(name: string, options: any) {
+    // TODO: This is where we would just generate function JSON from SLS object
+    // using getFunctionSlsObject(name, options). Currently defaulting to http in and out
+    return JSON.stringify(httpBinding);
+  }
+
+  public static getFunctionSlsObject(name: string, options: any) {
+    return FuncPluginUtils.defaultFunctionSlsObject(name);
+  }
+
+  private static defaultFunctionSlsObject(name: string) {
+    return {
+      handler: `${name}/index.handler`,
```

Comment:
we don't want `${name}/index.handler`, this will generate incorrect binding and function won't work.  

just `index.handler`

Reply:
This is the structure that `serverless.yml` needs to find the handler since it will live inside the directory `{functionName` and inside the file `index.js`, with the function name `handler`

Reply:
yes but the plugin will generate incorrect `scriptFilePath` since `function.json` now resides at the same level as `index.json`